Feature/add book functionality

chumakostya opened the pull request.

chumakostya committed:
Create DbInitializer for Categories table, rename and move BloggingContextFactory into a separate file, add Save method to interface IRepo, add BookController, add AddBook view

chumakostya committed:
Merge branch 'develop' into feature/Add_Book_Functionality

# Conflicts:
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Startup.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Views/AddBook/Index.cshtml
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Views/Shared/_Layout.cshtml
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Views/_ViewImports.cshtml

chumakostya committed:
1) Add dynamically added/removed input fields functionality
2)Add validation
3)Edit BookController and BookViewModel
4)Add AuthorViewModel
5)Add default image for book in wwwroot

chumakostya committed:
Removed hardcoded text for labels for authors

chumakostya committed:
Merge branch 'develop' into feature/Add_Book_Functionality

# Conflicts:
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Context/BookStoreContext.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170711134917_BookStore.Designer.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170711134917_BookStore.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170725194021_initial.Designer.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170725194021_initial.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170728155307_bookstore1.Designer.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170728155307_bookstore1.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/BookStoreContextModelSnapshot.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Repo/IRepo.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/LvivCompany.Bookstore.Web.csproj
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Startup.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Views/AddBook/AddBook.cshtml
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/appsettings.json

chumakostya committed:
Edit code to make possible project build with asynchronous methods

chumakostya committed:
Add generic repository base class Repo and make corresponding modifications in IRepo interface, derived repository classes and bookcontroller

chumakostya committed:
Modify code so that only one save is needed to save a book and its authors

chumakostya committed:
Remove redundant code when mapping book.CategoryId and model.CategoryId

IUnknownPtr commented on LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Views/Shared/_Layout.cshtml:
```diff
@@ -112,18 +96,17 @@
-
+        </div>
+    @RenderSection("scripts", required: false)
+    @RenderSection("styles", required: false)
```
Section for styles should be in `head`

IUnknownPtr commented on LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Context/BookStoreContext.cs:
```diff
@@ -24,9 +17,9 @@ public BookStoreContext(DbContextOptions<BookStoreContext> options) : base(optio
-
-
-       
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
```
Is this override needed?

IUnknownPtr commented on LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Controllers/BookController.cs:
```diff
@@ -0,0 +1,107 @@
+            var categoryList = await repoCategory.GetAllAsync();
+            model.Categories = categoryList.Select(c => new SelectListItem
+            {
+                Text = $"{c.Name}",
```
string interpolation is redundant

IUnknownPtr commented on LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Views/Book/AddBook.cshtml:
```diff
@@ -0,0 +1,219 @@
+﻿@model BookViewModel
```
This code looks ugly and it's wrong. Refactor to use only one `for` block.

chumakostya committed:
Make changes according to pull request commits

chumakostya committed:
Merge branch 'develop' into feature/Add_Book_Functionality

# Conflicts:
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Context/BookStoreContext.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Entities/Migrations/20170719123704_Initial.Designer.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Entities/Migrations/20170719123704_Initial.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Entities/Migrations/ApplicationContextModelSnapshot.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Controllers/HomeController.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Startup.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Views/Shared/_Layout.cshtml
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/appsettings.json

chumakostya committed:
Remove package references and edit async actions that cause warnings, remove duplicate from Startup class
Merge branch 'develop' into feature/Add_Book_Functionality

# Conflicts:
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170728155307_bookstore1.Designer.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170728155307_bookstore1.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170802145719_initial.Designer.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170802145719_initial.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170804092353_bookstore1.Designer.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Migrations/20170804092353_bookstore1.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Startup.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/ViewModels/BookViewModel.cs
#	LvivCompany.Bookstore/LvivCompany.Bookstore.Web/appsettings.json

chumakostya committed:
1) Move FullName property from Author model to AuthorViewModel
2) Apply IMapper to AddBook functionality
3) Join BookController and BookDetailController, BookDetailMapper and BookMapper, BookDetailViewModel, BookInfo and BookViewModel

chumakostya committed:
Move the code from ImageController to a separate class and implement add image functionality for the add book page

chumakostya committed:
Delete unnecessary image file

chumakostya committed:
Delete unnecessary folder Images and impose restrictions on book cover's image dimensions for add book page
